tasks/omnibus.py

```diff
@@ -118,12 +119,30 @@ def get_omnibus_env(
     if go_mod_cache:
         env['OMNIBUS_GOMODCACHE'] = go_mod_cache
 
-    env_override = ['INTEGRATIONS_CORE_VERSION', 'OMNIBUS_RUBY_VERSION']
-    for key in env_override:
+    external_repos = {
+        "INTEGRATIONS_CORE_VERSION": "https://github.com/DataDog/integrations-core.git",
+        "OMNIBUS_RUBY_VERSION": "https://github.com/DataDog/omnibus-ruby.git",
+    }
+    for key, url in external_repos.items():
         value = os.environ.get(key)
         # Only overrides the env var if the value is a non-empty string.
         if value:
             env[key] = value
+        ref = env[key]
+        if not re.fullmatch(r"[0-9a-f]{4,40}", ref):  # resolve only "moving" refs, such as `own/branch`
+            candidates = tuple(
+                line.split() for line in subprocess.check_output(("git", "ls-remote", url, ref), text=True).splitlines()
```

Comment:
For consistency we should probably use `ctx.run` to manage subprocesses. Also, adding [--refs](https://git-scm.com/docs/git-ls-remote#_output) looks somewhat safer.

```suggestion
                line.split() for line in ctx.run(f'git ls-remote --refs "{url}" "ref"').stdout.splitlines()
```

Reply:
Hum, seems I shouldn't have moved to `ctx.run`:
- #40227